games-action/ddnet-18.1 fails to compile: jobs.cpp:108:58: error: find is not a member of std (gentoo GCC-14)

This is likely something introduced by the recent gcc 14 maybe?

https://bugs.gentoo.org/927880

Build log https://927880.bugs.gentoo.org/attachment.cgi?id=888645

```
FAILED: CMakeFiles/engine-shared.dir/src/engine/shared/jobs.cpp.o 
/usr/bin/x86_64-pc-linux-gnu-g++ -DCONF_INFORM_UPDATE -DCONF_OPENSSL -DCONF_VIDEORECORDER -DGAME_RELEASE_VERSION=\"18.1\" -DGLEW_STATIC -I/var/tmp/portage/games-action/ddnet-18.1/work/ddnet-18.1_build/src -I/var/tmp/portage/games-action/ddnet-18.1/work/ddnet-18.1/src -I/var/tmp/portage/games-action/ddnet-18.1/work/ddnet-18.1/src/rust-bridge  -O2 -march=x86-64 -pipe -pipe -frecord-gcc-switches -fno-diagnostics-color -fmessage-length=0 -std=c++17 -fdiagnostics-color=always -fno-exceptions -Wall -Wextra -Wno-psabi -Wno-unused-parameter -Wno-missing-field-initializers -Wformat=2 -Wno-nullability-completeness -Wduplicated-cond -Wduplicated-branches -Wlogical-op -Wrestrict -Wshadow=global -Wsuggest-override -Wclass-memaccess -MD -MT CMakeFiles/engine-shared.dir/src/engine/shared/jobs.cpp.o -MF CMakeFiles/engine-shared.dir/src/engine/shared/jobs.cpp.o.d -o CMakeFiles/engine-shared.dir/src/engine/shared/jobs.cpp.o -c /var/tmp/portage/games-action/ddnet-18.1/work/ddnet-18.1/src/engine/shared/jobs.cpp
/var/tmp/portage/games-action/ddnet-18.1/work/ddnet-18.1/src/engine/shared/jobs.cpp: In member function ‘void CJobPool::RunLoop()’:
/var/tmp/portage/games-action/ddnet-18.1/work/ddnet-18.1/src/engine/shared/jobs.cpp:108:58: error: ‘find’ is not a member of ‘std’; did you mean ‘bind’?
  108 |                                 m_RunningJobs.erase(std::find(m_RunningJobs.begin(), m_RunningJobs.end(), pJob));
      |                                                          ^~~~
      |                                                          bind
At global scope:
cc1plus: note: unrecognized command-line option ‘-Wno-nullability-completeness’ may have been intended to silence earlier diagnostics
[46/304] /usr/bin/x86_64-pc-linux-gnu-g++ -DCONF_INFORM_UPDATE -DCONF_OPENSSL -DCONF_VIDEORECORDER -DGAME_RELEASE_VERSION=\"18.1\" -DGLEW_STATIC -I/var/tmp/portage/games-action/ddnet-18.1/work/ddnet-18.1_build/src -I/var/tmp/portage/games-action/ddnet-18.1/work/ddnet-18.1/src -I/var/tmp/portage/games-action/ddnet-18.1/work/ddnet-18.1/src/rust-bridge  -O2 -march=x86-64 -pipe -pipe -frecord-gcc-switches -fno-diagnostics-color -fmessage-length=0 -std=c++17 -fdiagnostics-color=always -fno-exceptions -Wall -Wextra -Wno-psabi -Wno-unused-parameter -Wno-missing-field-initializers -Wformat=2 -Wno-nullability-completeness -Wduplicated-cond -Wduplicated-branches -Wlogical-op -Wrestrict -Wshadow=global -Wsuggest-override -Wclass-memaccess -MD -MT CMakeFiles/engine-shared.dir/src/engine/shared/http.cpp.o -MF CMakeFiles/engine-shared.dir/src/engine/shared/http.cpp.o.d -o CMakeFiles/engine-shared.dir/src/engine/shared/http.cpp.o -c /var/tmp/portage/games-action/ddnet-18.1/work/ddnet-18.1/src/engine/shared/http.cpp
[48/304] /usr/bin/x86_64-pc-linux-gnu-g++ -DCONF_INFORM_UPDATE -DCONF_OPENSSL -DCONF_VIDEORECORDER -DGAME_RELEASE_VERSION=\"18.1\" -DGLEW_STATIC -I/var/tmp/portage/games-action/ddnet-18.1/work/ddnet-18.1_build/src -I/var/tmp/portage/games-action/ddnet-18.1/work/ddnet-18.1/src -I/var/tmp/portage/games-action/ddnet-18.1/work/ddnet-18.1/src/rust-bridge  -O2 -march=x86-64 -pipe -pipe -frecord-gcc-switches -fno-diagnostics-color -fmessage-length=0 -std=c++17 -fdiagnostics-color=always -fno-exceptions -Wall -Wextra -Wno-psabi -Wno-unused-parameter -Wno-missing-field-initializers -Wformat=2 -Wno-nullability-completeness -Wduplicated-cond -Wduplicated-branches -Wlogical-op -Wrestrict -Wshadow=global -Wsuggest-override -Wclass-memaccess -MD -MT CMakeFiles/engine-shared.dir/src/engine/shared/config.cpp.o -MF CMakeFiles/engine-shared.dir/src/engine/shared/config.cpp.o.d -o CMakeFiles/engine-shared.dir/src/engine/shared/config.cpp.o -c /var/tmp/portage/games-action/ddnet-18.1/work/ddnet-18.1/src/engine/shared/config.cpp
ninja: build stopped: subcommand failed.
 * ERROR: games-action/ddnet-18.1::guru failed (compile phase):
 *   ninja -v -j16 -l0 failed
 *
```

--- a/src/engine/shared/jobs.cpp
+++ b/src/engine/shared/jobs.cpp
@@ -1,6 +1,7 @@
 /* (c) Magnus Auvinen. See licence.txt in the root of the distribution for more information. */
 /* If you are missing that file, acquire a complete release at teeworlds.com.                */
 #include "jobs.h"
+#include <algorithm>
 
 IJob::IJob() :
 	m_pNext(nullptr),

